File: clients/dart/cldf/lib/models/enums/media_designation.dart
import 'package:json_annotation/json_annotation.dart';

/// Purpose or type of media content
@JsonEnum(valueField: 'value')
enum MediaDesignation {
  topo('topo'), // Route diagram/map
  beta('beta'), // How-to information
  approach('approach'), // Access/approach info
  log('log'), // Climb documentation
  overview('overview'), // General view/panorama
  conditions('conditions'), // Current conditions
  gear('gear'), // Gear placement/requirements
  descent('descent'), // Descent information
  other('other'); // Unspecified purpose

  const MediaDesignation(this.value);
  final String value;

  @override
  String toString() => value;

  /// Parse from string value, defaulting to 'other' if unknown
  static MediaDesignation fromValue(String? value) {
    if (value == null) return MediaDesignation.other;
    return MediaDesignation.values.firstWhere(
      (d) => d.value == value.toLowerCase(),
      orElse: () => MediaDesignation.other,
    );
  }
}

File: clients/java/cldf-java/src/main/java/app/crushlog/cldf/models/enums/MediaDesignation.java
package app.crushlog.cldf.models.enums;

import java.util.Arrays;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** Purpose or type of media content */
public enum MediaDesignation {
  TOPO("topo"), // Route diagram/map
  BETA("beta"), // How-to information
  APPROACH("approach"), // Access/approach info
  LOG("log"), // Climb documentation
  OVERVIEW("overview"), // General view/panorama
  CONDITIONS("conditions"), // Current conditions
  GEAR("gear"), // Gear placement/requirements
  DESCENT("descent"), // Descent information
  OTHER("other"); // Unspecified purpose

  private final String value;

  MediaDesignation(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @JsonCreator
  public static MediaDesignation fromValue(String value) {
    if (value == null) {
      return OTHER; // Default to OTHER if null
    }
    return Arrays.stream(MediaDesignation.values())
        .filter(designation -> designation.value.equalsIgnoreCase(value))
        .findFirst()
        .orElse(OTHER); // Default to OTHER if not found
  }

  @Override
  public String toString() {
    return value;
  }
}
